polyply/tests/test_topology.py

```diff
@@ -238,3 +239,206 @@ def test_convert_nonbond_to_sig_eps():
+        C       C       1       0.1335  502080.0
+        [ moleculetype ]
+        test 3
+        [ atoms ]
+        1 C   1 test C1 1   0.0 14.0
+        2 C   1 test C2 2   0.0 12.0
+        [ bonds ]
+        1 2  1
+        [ system ]
+        some title
+        [ molecules ]
+        test 1
+        """,
+        {"bonds": [Interaction(atoms=(0, 1), parameters=["1", "0.1335", "502080.0"], meta={})]}
+        ),
+        # test three element define
+        (
+        """
+        [ defaults ]
+        1.0   1.0   yes  1.0     1.0
+        [ angletypes ]
+        CE1   CE1	CT2	5	123.50	401.664	0.0	0.0
+        [ moleculetype ]
+        test 3
+        [ atoms ]
+        1 CE1   1 test C1 1   0.0 14.0
+        2 CE1   1 test C2 2   0.0 12.0
+        3 CT2   1 test C3 3   0.0 12.0
+        [ angles ]
+        1 2  3 1
+        [ system ]
+        some title
+        [ molecules ]
+        test 1
+        """,
+        {"angles": [Interaction(atoms=(0, 1, 2), parameters=["5", "123.50", "401.664", "0.0", "0.0"],
+                                meta={})]}
+        ),
+        # test reverse match
+        (
+        """
+        [ defaults ]
+        1.0   1.0   yes  1.0     1.0
+        [ angletypes ]
+        CE1    CE2	CT2	5	123.50	401.664	0.0	0.0
+        [ moleculetype ]
+        test 3
+        [ atoms ]
+        1 CT2   1 test C1 1   0.0 14.0
+        2 CE2   1 test C2 2   0.0 12.0
+        3 CE1   1 test C3 3   0.0 12.0
+        [ angles ]
+        1  2  3 1
+        [ system ]
+        some title
+        [ molecules ]
+        test 1
+        """,
+        {"angles": [Interaction(atoms=(0, 1, 2), parameters=["5", "123.50", "401.664", "0.0", "0.0"],
+                                meta={})]}
+        ),
+        # test generic match
+        (
+        """
+        [ defaults ]
+        1.0   1.0   yes  1.0     1.0
+        [ dihedraltypes ]
+        X  CE2  CE1  X    5   123.50	401.664	0.0	0.0
+        X  CT2  CE1  X    5   123.50	401.664	0.0	0.0
+        [ moleculetype ]
+        test 3
+        [ atoms ]
+        1 CT2   1 test C1 1   0.0 14.0
+        2 CE2   1 test C2 2   0.0 12.0
+        3 CE1   1 test C3 3   0.0 12.0
+        4 CT2   1 test C4 4   0.0 12.0
+        5 CT2   1 test C5 5   0.0 14.0
+        [ dihedrals ]
+        1  2  3  4 1
+        2  3  4  5 1
+        [ system ]
+        some title
+        [ molecules ]
+        test 1
+        """,
+        {"dihedrals": [Interaction(atoms=(0, 1, 2, 3), parameters=["5", "123.50", "401.664", "0.0", "0.0"],
+                                meta={}),
+                       Interaction(atoms=(1, 2, 3, 4), parameters=["5", "123.50", "401.664", "0.0", "0.0"],
+                                meta={})]}
+        ),
+        # test generic improper
+        (
+        """
+        [ defaults ]
+        1.0   1.0   yes  1.0     1.0
+        [ dihedraltypes ]
+        CT2   X     X     CT2    5   123.50	401.664	0.0	0.0
+        CT2   X     X     CE2    5   123.50	401.664	0.0	0.0
```

Comment:
Give htem different param values.
Add a non-matching pattern. Add a pattern that cannot match, e.g. X X X X ...
Add a matching pattern without X's. Which should be used? Which is used?

Reply:
>  Add a matching pattern without X's. Which should be used? Which is used?< 

GROMACS goes by priority or position in the itp file. There is no documentation for this.

Reply:
so gromacs takes the more specific interaction if defined.